en/services/payload_control.md

```diff
@@ -0,0 +1,62 @@
+
+> **Tip** Payload specific commands should be used (where supported) in preference to [arbitrary payload commands](#arbitraryunknown-payload-commands) as generally they provide more "tailored" control over the payload, a better GCS user experience, and more informative logs.
+
+
+Message | Description
+-- | --
+<a id="MAV_CMD_DO_GRIPPER"></a>[MAV_CMD_DO_GRIPPER](../messages/common.md#MAV_CMD_DO_GRIPPER) | Command to engage and release a gripper.
+<a id="MAV_CMD_DO_WINCH"></a>[MAV_CMD_DO_WINCH](../messages/common.md#MAV_CMD_DO_WINCH) | Command to operate a specified winch.
+<a id="MAV_CMD_ILLUMINATOR_ON_OFF"></a>[MAV_CMD_ILLUMINATOR_ON_OFF](../messages/common.md#MAV_CMD_ILLUMINATOR_ON_OFF) | Command to turn illuminators ON/OFF. An illuminator is a light source that for lighting up dark areas external to the sytstem, such as a torch or searchlight.
+<a id="MAV_CMD_DO_SET_CAM_TRIGG_INTERVAL"></a>[MAV_CMD_DO_SET_CAM_TRIGG_INTERVAL](../messages/common.md#MAV_CMD_DO_SET_CAM_TRIGG_INTERVAL) | Command to set time interval between camera captures
+<a id="MAV_CMD_DO_SET_CAM_TRIGG_DIST"></a>[MAV_CMD_DO_SET_CAM_TRIGG_DIST](../messages/common.md#MAV_CMD_DO_SET_CAM_TRIGG_DIST) | Command to set distance travelled between camera captures.
+
+
+## Arbitrary/Unknown Payload Commands
+
+The [MAV_CMD_DO_SET_ACTUATOR](#MAV_CMD_DO_SET_ACTUATOR) and [MAV_CMD_DO_SET_SERVO](#MAV_CMD_DO_SET_SERVO) commands allow setting the value of flight controller outputs, without having to know what that specific hardware is. 
+
+> **Tip** Use the [commands for known payload types](#payload-specific-commands) where possible.
+  These are more intuitive for users, and in logs.
+
+`MAV_CMD_DO_SET_ACTUATOR` is better designed and more recent than `MAV_CMD_DO_SET_SERVO`, but has not yet been widely adopted.
+- `DO_SET_ACTUATOR` can in theory be used on any bus - MAIN, AUX, UAVCAN, etc., while `DO_SET_SERVO` can only be used on PWM busses, and is generally hardcoded to map instance numbers to AUX outputs (this is not mandated by the specification, so cannot be relied on).
+- `DO_SET_ACTUATOR` can set multiple outputs in a single message. 
+
+Message | Description
+-- | --
+<a id="MAV_CMD_DO_SET_ACTUATOR"></a>[MAV_CMD_DO_SET_ACTUATOR](../messages/common.md#MAV_CMD_DO_SET_ACTUATOR) | Sets actuators (e.g. servos) to a desired value. The actuator numbers are mapped to specific outputs (e.g. on any MAIN or AUX PWM or UAVCAN) using a flight-stack specific mechanism (for example, parameters).
+<a id="MAV_CMD_DO_SET_SERVO"></a>[MAV_CMD_DO_SET_SERVO](../messages/common.md#MAV_CMD_DO_SET_SERVO) | Sets a servo, identified by a specified instance number, to a specified PWM value.
+
+
+> **Note** **Implementations:**
+  - `MAV_CMD_DO_SET_SERVO` is impemented on both ArduPilot and PX4.
+    In both cases instance numbers map to AUX outputs.
+  - `MAV_CMD_DO_SET_ACTUATOR` is impemented only on PX4 (June 2021).
```

Comment:
@bkueng My understanding is that the flight stack provides the mapping between the command inputs (param1, param2, param3 only for PX4) and the outputs - say AUX1, AUX4, UAVCAN3 using whatever method it likes.

On PX4 the docs now confuse me: https://docs.px4.io/master/en/payloads/#mission-triggering

My assumption was that RC_MAP_AUX1 to RC_MAP_AUX3 take values that indicate the outputs reference by param1, param2, param3. So for example you might in some magical world enter in `RC_MAP_AUX1` a value that means UAVCAN4 or MAIN5.

However on re-read I don't think it works that way. I THINK that what happens is that the params 1 to 3 map to the AUX1- AUX3: feed-through channels - i.e. always the same outputs as defined by the current mixer.
I guess that's good, because you can map the AUX1-3 to UAVCAN if you like or MAINx-x+3. But it isn't as flexible as letting people assign outputs via params in any way they like. 

Anyway, can you confirm my understanding - the mapping on PX4 has nothing to do with the RC_MAP_AUXx params other than the fact that they map to the same mixer outputs?

Reply:
Thanks @bkueng I have modified both this and PX4 to remove ambiguity/inaccuracy.